core/capabilities/remote/target/client.go

```diff
@@ -146,11 +162,9 @@ func (c *client) Receive(msg *types.MessageBody) {
 		return
 	}
 
-	go func() {
-		if err := req.OnMessage(ctx, msg); err != nil {
-			c.lggr.Errorw("failed to add response to request", "messageID", messageID, "err", err)
-		}
-	}()
+	if err := req.OnMessage(ctx, msg); err != nil {
```

Comment:
Why is it OK to call this in the same goroutine now?

Reply:
this will be ok once ks-327 is done, being worked on at the moment